frontend/packages/noobaa-storage-plugin/src/components/data-consumption-card/data-consumption-card.tsx

```diff
@@ -81,6 +82,73 @@ const DataConsumptionCard: React.FC<DashboardItemProps> = ({
     if (suffixLabel) suffixLabel = `(in ${suffixLabel})`;
   }
 
+  if (isLoading) {
+    body = (
+      <>
+        <div className="skeleton-text nb-data-consumption-card__chart-skeleton" />
+        <GraphEmpty height={200} loading />
+        <div className="skeleton-text nb-data-consumption-card__chart-legend-skeleton" />
+      </>
+    );
+  } else if (!error && !chartData.some(_.isEmpty)) {
+    body = (
+      <>
+        <div className="nb-data-consumption-card__chart-label text-secondary">
+          {CHART_LABELS[sortByKpi]} {suffixLabel}
+        </div>
+        <Chart
+          containerComponent={
+            <ChartVoronoiContainer
+              labelComponent={<ChartTooltip style={{ fontSize: 8, paddingBottom: 0 }} />}
+              labels={({ datum }) => `${datum.y} ${maxUnit}`}
+              voronoiDimension="x"
+            />
+          }
+          minDomain={{ y: 0 }}
+          maxDomain={{ y: maxVal + 10 }}
+          domainPadding={{ x: [padding, padding] }}
+          legendComponent={
+            <ChartLegend
+              themeColor={ChartThemeColor.purple}
+              data={legendData}
+              orientation="horizontal"
+              symbolSpacer={5}
+              gutter={2}
+              height={30}
+              padding={{ top: 50, bottom: 0 }}
+              style={{ labels: { fontSize: 8 } }}
+            />
+          }
+          legendPosition="bottom-left"
+          padding={{
+            bottom: 50,
+            left: 30,
+            right: 20,
+            top: 30,
+          }}
+          themeColor={ChartThemeColor.purple}
+        >
+          <ChartAxis style={{ tickLabels: { fontSize: 8, padding: 2 } }} />
+          <ChartAxis
+            dependentAxis
+            showGrid
+            tickCount={10}
+            style={{
+              tickLabels: { fontSize: 8, padding: 0 },
+            }}
+          />
+          <ChartGroup offset={sortByKpi === BY_EGRESS ? 0 : 11}>
+            {chartData.map((data, i) => (
+              <ChartBar key={i} data={data} /> // eslint-disable-line react/no-array-index-key
+            ))}
+          </ChartGroup>
+        </Chart>
+      </>
+    );
+  } else {
+    body = <GraphEmpty />;
```

Comment:
put this as initial value of body

Reply:
We want to show loading states first and this is rendered in case of errors.

Reply:
I see no problem with it. If the initial value is `graphyEmpty`. And if it is loading then that should get changed.

Reply:
I would not prefer overriding :)

Reply:
Technically you are still overriding with the else statement. This would result in you not requiring `else` statement.

Reply:
Think about it, I dont think so :)

Reply:
Maybe I am missing something. If you say so, LGTM. :)